code/datums/components/supermatter_crystal.dm

```diff
@@ -218,11 +220,34 @@
 /datum/component/supermatter_crystal/proc/intercept_z_fall(datum/source, list/falling_movables, levels)
 	SIGNAL_HANDLER
 	for(var/atom/movable/hit_object as anything in falling_movables)
-		if(hit_object == source)
-			continue
+		if(parent == hit_object)
+			return
+
 		bumped_hit(parent, hit_object)
 	return FALL_INTERCEPTED | FALL_NO_MESSAGE
 
+/datum/component/supermatter_crystal/proc/on_z_impact(datum/source, turf/impacted_turf, levels)
+	SIGNAL_HANDLER
+
+	for(var/atom/movable/hit_object as anything in impacted_turf)
+		if(parent == hit_object)
+			return
+
+		var/mob/living/poor_target = locate(/mob/living) in impacted_turf
```

Comment:
Why do you iterate over all movables in the turf just to do a `locate(mob) in`

Reply:
Well, it didn't work properly from what I remember, so I did this, I'll recheck it later.

Reply:
`for(var/mob/living/guy in impacted_turf)`?

Reply:
Thanks